components/fleetArchiveDS.js

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { Datasource } = require("./datasource.js");
+const logging = require("./logging.js");
+const _ = require("lodash");
+
+class FleetArchiveLogs extends Datasource {
```

Comment:
There's a bunch of logic in here that really belongs in fleetarchive it self (ie the work figuring out which logs are relevant given a trip_id or vehicle_id).  

fleetarchive can use SQL ... that's going to be faster, simpler, better tested.

I also claim that other users of fleetarchive will want/need it as well.

Reply:
That would be very helpful. I guess this PR is still needed for now before that work happens?